src/rosdiscover/observer/ros1.py

```diff
@@ -94,3 +100,44 @@ def _transform_state_to_nodeinfo(self, state: SystemState) -> Collection[NodeInf
                         reorganized_nodes[node_name].provides.add(service)
 
         return list(reorganized_nodes.values())
+
+    def launch_from_config(self, sleep_time: float) -> t.Sequence[Popen]:
+        # Eagerly check launch files exist - don't start any if one doesn't exist
+        app_instance = self._app_instance
+        for launch in self._config.launches:
+            if not app_instance.files.exists(launch.filename):
+                raise FileNotFoundError(launch.filename)
+
+        processes = []
+
+        for launch in self._config.launches:
+            launch_script = self._generate_launch_script_on_host(launch)
+            cmd = f"bash {launch_script}"
```

Comment:
prefer /bin/bash to be a bit more robust against bad PATH variables

Reply:
Done